pkg/controller/resourcequota/resource_quota_monitor.go

```diff
@@ -167,14 +168,21 @@ func (qm *QuotaMonitor) controllerFor(ctx context.Context, resource schema.Group
 	shared, err := qm.informerFactory.ForResource(resource)
 	if err == nil {
 		logger.V(4).Info("QuotaMonitor using a shared informer", "resource", resource.String())
-		shared.Informer().AddEventHandlerWithResyncPeriod(handlers, qm.resyncPeriod())
-		return shared.Informer().GetController(), nil
+		registration, err := shared.Informer().AddEventHandlerWithResyncPeriod(handlers, qm.resyncPeriod())
+		if err != nil {
+			logger.Error(err, "QuotaMonitor failed to add event handler to shared informer", "resource", resource.String())
+			return nil, nil, err
+		}
+		return shared.Informer().GetController(), func() {
+			logger.V(4).Info("QuotaMonitor removed resource", "resource", resource.String())
+			shared.Informer().RemoveEventHandler(registration)
```

Comment:
This removes the event handler but doesn't do anything to clear out things resource quota recorded from the resource... what happens to the `resourceChanges` queue and anything the quota controller recorded for use of this resource?

There are two cases I think we need to handle differently:
* Removal of a CRD: if resource quota is watching widgets v1beta1, and I delete the widgets v1beta1 API, usage should probably go to 0?
* Addition of a new version to a CRD: if resource quota is watching widgets v1beta1, and I update the CRD to define a v1 version, that will show up as "start watching v1, stop watching v1beta1". In that case, I think we would want to transition resourcequota calculations to the v1 API, not clear them to 0 and then recalculate them against v1.

We need test coverage for both of those cases to make sure we do the right thing.